src/paintcomponents/data/DataToPoint.java

```diff
@@ -53,7 +53,7 @@ public DataToPoint(int x, int y, Type expectedType) {
 	public Object fetchData() throws NoConnectingLineSegmentException, NoSuchElementException, DataFromPointNoDataProviderException, DataFromPointProviderCannotProvideDataException{
 		if(this.lineSegment == null) throw new NoConnectingLineSegmentException();
 		
-		Object returnVal = lineSegment.getFromPoint().getData();
+		Object returnVal = lineSegment.getFromPoint().getData().toString();
```

Comment:
This usage is incorrect, please change it back to getData() since we will check the incoming instance's data type by calling its getClass method.